src/index.js

```diff
@@ -151,31 +151,25 @@ function check(autoCheckElement: AutoCheckElement) {
       if (response.status !== 200) {
         throw new ErrorWithResponse(response.statusText, response)
       }
-      return response.text()
+      return response
     })
-    .then(message => {
+    .then(response => {
       autoCheckElement.dispatchEvent(new CustomEvent('load'))
       if (autoCheckElement.required) {
         input.setCustomValidity('')
       }
-      input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {message}, bubbles: true}))
+      input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {response: response.clone()}, bubbles: true}))
 
       // Mark the component as not being in-flight any more.
       abortControllers.delete(autoCheckElement)
     })
-    .catch(async error => {
-      const message = await error.response.text()
-      const contentType = error.response.headers.get('Content-Type')
-
+    .catch(error => {
```

Comment:
This can catch both network errors thrown from `fetch` and the custom `ErrorWithResponse` thrown from the `then` block. So the `error.response` reference below will throw an exception when dealing with a network error object.

Can we switch to `async/await` here and separate the network error catch block? I think that will let us delete the `ErrorWithResponse` class.

Reply:
Initially I was scared that switching to `async/await` would make it a lot harder to review this PR so I left it out. I pushed commits that do that now and if it's hard to read I can split it out into a different PR, just let me know.